packages/block-library/src/post-categories/edit.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+function PostCategoriesDisplay() {
+	const [ categories ] = useEntityProp( 'postType', 'post', 'categories' );
+	const categoryLinks = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( 'core' );
+			let loaded = true;
+			const links = categories.map( ( categoryId ) => {
+				const category = getEntityRecord('taxonomy', 'category', categoryId );
+				if ( ! category ) {
+					return ( loaded = false );
+				}
+				return (
+					<a key={categoryId } href={ category.link }>
+						{ category.name }
+					</a>
+				);
+			} );
+			return loaded && links;
+		},
+		[ categories ]
+	);
+	return (
+		categoryLinks &&
+		( categoryLinks.length === 0
+			? __( 'No categories.' )
+			: categoryLinks.reduce( ( prev, curr ) => [ prev, ', ', curr ] ) )
+	);
+}
+
+export default function PostCategoriesEdit() {
+	if ( ! useEntityId( 'postType', 'post' ) ) {
```

Comment:
With the introduction of block context, the client-side implementation should also reference the context values as well, rather than via `useEntityId` and `useEntityProp`. At least in getting the `postId` and `postType`, it should largely amount to using `context` as a property of the edit function.

```js
export default function PostCategoriesEdit( { context } ) {
    const { postId, postType } = context;
}
```

(`postType` will need to be listed in `block.json` `context` alongside the existing `postId`)

To get the categories, it's a little more work using [`core-data` entities](https://github.com/WordPress/gutenberg/blob/master/packages/core-data/README.md) to get the post's categories:

```js
const categories = useSelect( ( select ) => 
    select( 'core' ).getEditedEntityRecord( 'postType', postType, postId )?.categories )
);
```

See also for reference example: #22359 (specifically, https://github.com/WordPress/gutenberg/pull/22359#issuecomment-628797161)

Related: #21665

Reply:
I have not been able to figure out how to get all taxonomies from core-data, or a named custom taxonomy?

Reply:
I have a list of all hierarchical taxonomies, but I don't know if I can use this core-data to check if a post uses that taxonomy.